_blog-src/_posts/2016-01-26-export-import.markdown

```diff
@@ -0,0 +1,155 @@
+
+The three main use cases that the export/import feature supports are:
+
+- Backing up all your apiman data
+- Upgrading to a newer version of apiman
+- Migrating apiman data from a Test environment to a Production environment
+
+The export/import operations are only available to Administrative users. The export/import feature is accessed through the admin operations menu:
+
+![Image: Admin Choices](/blog/images/2016-01-26/export_import_1.png)
+
+Once you select the **Export/Import** Data feature, this menu is displayed:
+
+![Image: Export/Import Menu](/blog/images/2016-01-26/export_import_2.png)
+
+One thing to keep in mind is that while you are importing or exporting data, no changes to data should be made or else the export/import may encounter an error, or may result in incomplete results. To be safe, you should disable user access to the API Manager, both the Management UI and its REST interface, for the duration of the import or export operation.
+
+We'll look at exporting data first.
+
+### Backing up apiman Data
+
+To make a backup of all your apiman data, simply select the "Export All" button. The apiman data will be written to a file and downloaded by your browser. Your browser's settings will determine where the file is saved.
+
+The apiman data is written to a file named: api-manager-export.json
+
+As its name indicates, the apiman data is written in JSON form. This format provides us with  several advantages. First of all, it's the format in which apiman is able to import data. (We'll perform an import later in this post.) Secondly, it's a text file where the contents of the file are human readable. The content of the file is ALL the apiman data, both the data elements that you have created and the data elements with which apiman is preconfigured.
+
+Reading this file can greatly increase your understanding of the elements that are defined in apiman. Let's take a look at the elements in the file:
+
+- **Users** - The preconfigured "admin" user is defined here, as are the new users we create.
+- **Gateways** - The preconfigured apiman Gateway is defined here.
+- **Roles** - The preconfigured, permission-based roles, and new roles that we create, are defined here. For example, the "OrganizationOwner" role is shown to have these permissions: [ "apiAdmin", "orgAdmin", "apiView", "orgEdit", "clientEdit", "clientAdmin", "planView", "orgView", "planAdmin", "clientView", "planEdit", "apiEdit" ]
+- **Policy Definitions** -  Next, the preconfigured policies, and new policies t2015-11-12-micro-services.markdownhat we create, are defined here. For example: the "RateLimitingPolicy" is described as "Enforces rate configurable request rate limits on an API.  This ensures that consumers can't overload an API with too many requests."
+- The remainder of the file includes the elements that we create: **Organizations, Plans, APIs, and Client Apps**. For example, here is the definition of the "echo" API that we created:
+
+	  "Apis" : [ {
```

Comment:
This syntax for code blocks works fine in asciidoctor, but it seems to trick it into think it's part of the list, which gives it a weird indent.

I suggest you flatten it against the edge and use the ``` syntax

e.g.

\`\`\`
code
\`\`\`

becomes

```
code
```